### PR TITLE
letsencrypt: Update Alpine, Python and Certbot

### PR DESCRIPTION
**Pull request by agners**
Update to Alpine 3.18 and Python version 3.11. Also update Certbot to the latest release 2.7.4 along with all DNS authenticator plug-ins.

### CODE REVIEW

File: letsencrypt/rootfs/etc/cont-init.d/file-structure.sh [hunk 1]
```diff
@@ -6,9 +6,7 @@ mkdir -p /data/workdir
 mkdir -p /data/letsencrypt
 
 # Setup Let's encrypt config
-echo -e "dns_cloudxns_api_key = $(bashio::config 'dns.cloudxns_api_key')\n" \
-      "dns_cloudxns_secret_key = $(bashio::config 'dns.cloudxns_secret_key')\n" \
-      "dns_digitalocean_token = $(bashio::config 'dns.digitalocean_token')\n" \
+echo -e "dns_digitalocean_token = $(bashio::config 'dns.digitalocean_token')\n" \
       "certbot_dns_directadmin:directadmin_url = $(bashio::config 'dns.directadmin_url')\n" \
```
Comment (sairon) [inline letsencrypt/rootfs/etc/cont-init.d/file-structure.sh:10]: Shouldn't this and the following also be changed from the `dist_name:plugin_name` just to `plugin_name`?

**Review by sairon — COMMENTED** [review]
Looks good, checked the Certbot release notes and haven't found any gotcha. Just one thing in comments.

File: letsencrypt/rootfs/etc/cont-init.d/file-structure.sh (hunk 1 above)
Comment (agners) [reply letsencrypt/rootfs/etc/cont-init.d/file-structure.sh]: Good catch! There are a couple of old style configs still, I've updated them all now :+1:

**Review by sairon — APPROVED** [review]
Looks good now :+1:

**Review by frenck — APPROVED** [review]

**Pull request merged**